mat/cholesky.go

```diff
@@ -340,6 +340,70 @@ func (c *Cholesky) Scale(f float64, orig *Cholesky) {
 	c.cond = orig.cond // Scaling by a positive constant does not change the condition number.
 }
 
+// AppendOneSym computes the Cholesky decomposition of the original matrix A
+// with the data appended to the original matrix A, storing the result into the
+// receiver. That is orig contains the Cholesky decomposition of A, and at completion
+// it contains the Cholesky decomposition of
+//  [A a]
+//  [a k]
+// where k = r[len(r)-1] and a = r[:len(r)-1]. In order for the updated matrix
+// to be positive definite, it must be the case that n > a' A^-1 a. If this
+// condition does not hold then AppendOneSym will return false and the receiver
+// will not be updated.
+//
+// AppendRow will panic if len(r) != orig.Size()+1 or if orig does not contain
+// a valid decomposition.
+func (chol *Cholesky) AppendOneSym(orig *Cholesky, r Vector) (ok bool) {
+	n := orig.Size()
+	if r.Len() != n+1 {
+		panic(badSliceLength)
+	}
+	if !orig.valid() {
+		panic(badCholesky)
+	}
+
+	// The algorithm is commented here, but see also
+	//  https://math.stackexchange.com/questions/955874/cholesky-factor-when-adding-a-row-and-column-to-already-factorized-matrix
+	// We have A and want to compute the Cholesky of
+	//  [A a]
+	//  [a k]
```

Comment:
Where is the k from? I don't see that in the post above, also update a's to v's (and make the lower v => v').

Reply:
The post above assumes that k = 1, but in general there's nothing in particular requiring that to be be true. The important part is that k is positive, or more accurately that  k - a' A a is positive (and given that a' A a is always positive, this implies that k has to be positive)